sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -139,6 +143,31 @@ case class CollectSet(
 
   def this(child: Expression) = this(child, 0, 0)
 
+  /* SPARK-31500
+   * Array[Byte](BinaryType) Scala equality don't works as expected
```

Comment:
nit: 
```
  /* 
   * SPARK-31500: Array[Byte](BinaryType) Scala equality don't works as expected
```

Reply:
It is not a Scala issue. Java byte arrays use referential equality and identity hash codes. This has tripped up many many people before.

Reply:
Yes, I know the main reason, I'm going tom explain it better